### ISSUE TITLE
Not working for MacOS or Windows if replacement has space

### ISSUE THREAD

**Issue by DzmVasileusky** [ALMaclaine/replace#5]
This working only for MacOS
`node ./node_modules/replace/bin/replace.js 'a b' 'b a' src/index.html`

This working only for Windows
`node ./node_modules/replace/bin/replace.js \"a b\" \"b a\" src/index.html`

How to get it work cross platform?

**Comment by ALMaclaine** [comment]
I'm somewhat unclear on what you're saying the issue is. At first I thought you were saying it doesn't work at all, then maybe I thought you were saying it doesn't work for replacing quoted strings.

I'm deving replace on a mac, but given it seems you're using a bash terminal for windows I'd think they'd work fairly consistently. There's no OS specific code anywhere in what I've written.

With that said, I cannot reproduce any issues on mac for any variation on your example.

Additionally, and I'm not a bash expert, but I don't believe 

```node ./node_modules/replace/bin/replace.js \"a b\" \"b a\" src/index.html```

is a valid command because you're escaping the quotations outside of a string. I'm not sure what the point of escaping quotations outside of a string would be and I doubt it works, it certainly doesn't work on my terminal

As first stated I'm slightly unclear what you're saying doesn't work, but my instinct is that you're attempting to replace quoted material inside your file and you aren't providing the appropriate match.

I'm stabbing in the dark here but maybe something like this works:

```node ./node_modules/replace/bin/replace.js "'a b'" "'b a'" src/index.html```
or
```node ./node_modules/replace/bin/replace.js \"a b\" \"b a\" src/index.html```

In conclusion, I couldn't reproduce anything close to what this issue is describing on mac, so I was not inclined to try on windows. If you still find this issue persists and can provide me more context I will look into it further.

**Comment by DzmVasileusky** [comment]
Sorry it seems I miss some important notes.
Both commands are inside strings: there are commands in package.json scripts section. 

index.html
`<!DOCTYPE html><html lang="en"><head lang="en"><meta charset="UTF-8"><base href="/app/"></head><body></body></html>`

package.json
`"local-setup": "node ./node_modules/replace/bin/replace.js '<base href=\"/app/\">' '<base href=\"/app\">' src/index.html"`

This is working perfectly on MacOS.
But on Windows it's throwing error:
```
> crexi@7.19.0 local-setup D:\Work\Projects\app
> ./node_modules/replace/bin/replace.js '<base href="/app/">' '<base href="/app">' src/index.html

The system cannot find the file specified.
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! app@7.19.0 local-setup: `./node_modules/replace/bin/replace.js '<base href="/app/">' '<base href="/app">' src/index.html`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the app@7.19.0 local-setup script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
```

Then I tried to find crossplatform solution.

`"local-setup": "node ./node_modules/replace/bin/replace.js \"<base href=\"/app/\">\" \"<base href=\"/app\">\" src/index.html"`
Is legal for both Win and Mac, but doesn't replace anything.

`"local-setup": "node ./node_modules/replace/bin/replace.js href=\\\"/app/\\\" href=\\\"/app\\\" src/index.html"`
Is legal and working but less accurate (I had to remove `<base ` part).

**Comment by ALMaclaine** [comment]
I'll look into this shortly.

**Comment by ALMaclaine** [comment]
Is this issue not because of file path differences between macOS and windows...?

https://stackoverflow.com/a/50998274
https://github.com/yarnpkg/yarn/issues/1729#issuecomment-261802770

**Comment by DzmVasileusky** [comment]
@ALMaclaine Maybe, do you have Win environment to reproduce it?

**Comment by ALMaclaine** [comment]
I'm sorry, but I do not believe this is an issue with replace, it seems to be an issue with windows or your set up.

https://stackoverflow.com/questions/39190164/npm-the-system-cannot-find-the-path-specified?rq=1

If you can demonstrate this is an issue with replace I will gladly fix it.

**Comment by iofluxdev1** [comment]
Same issue, not usable on windows.

replace 'xxxfoo' 'xxxbar' . -r
Invalid switch - .
No files replaced


replace 'xxxfoo' 'xxxbar' *
Invalid switch - *
No files replaced


replace 'xxx foo' 'xxx bar' *
Invalid switch - 'xxx
No files replaced

**Comment by ALMaclaine** [comment]
Can you double check you aren't using the windows built in replace command as referenced in this issue #7.

**Comment by iofluxdev1** [comment]
@ALMaclaine Thank you, that was the problem I had. Installing globally and then using the command line results in windows replace.exe being used instead of the global npm package.

**Comment by ALMaclaine** [comment]
@iofluxdev1 No problem!
